### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by MBkkt**
[Improve s2region_term_indexer:](https://github.com/google/s2geometry/pull/296/commits/da933587ab1f8c162d31462e6b3e2036fa72c9fe) 
* Add ability to reuse term buffer
* Add option for optimize index size if query only points

[Add access to coverer](https://github.com/google/s2geometry/commit/9c78b46b61f1a14de0349b1767fd7c9bfcbf5cae)

[Add access to S2Polygon::subregion_bound_](https://github.com/google/s2geometry/commit/e8d837549cc6c101b0745e8f7609e9ee60832382)

[Add S2_SYSTEM_HEADERS cmake options to avoid s2 compile time warnings from s2 headers](https://github.com/google/s2geometry/commit/e6b14e87c49ed8ce625758f67481b79f83954db1) 

[Add S2_INSTALL option with keeping old behavior](https://github.com/google/s2geometry/commit/e81eb6526f7b6e40eca0c04e067b86897cb439e4)

Needs tests run! PR now mostly for discussion.

### CODE REVIEW

File: src/s2/s2region_term_indexer.cc [hunk 1]
```diff
@@ -178,14 +193,20 @@ vector<string> S2RegionTermIndexer::GetIndexTermsForCanonicalCovering(
...
     S2_DCHECK_EQ(0, (level - options_.min_level()) % options_.level_mod());
+    // assume level <= options_.true_max_level()
 
-    if (level < true_max_level) {
-      // Add a covering term for this cell.
-      terms.push_back(GetTerm(TermType::COVERING, id, prefix));
-    }
-    if (level == true_max_level || !options_.optimize_for_space()) {
-      // Add an ancestor term for this cell at the constrained level.
-      terms.push_back(GetTerm(TermType::ANCESTOR, id.parent(level), prefix));
```
Comment (MBkkt) [inline src/s2/s2region_term_indexer.cc:188]: `id.parent(level) == parent` is always true here

**Comment by jmr** [comment]
Having 6 PRs in 1 will quickly become unwieldy. Please split up future PRs for easier disucssion.

**Comment by MBkkt** [comment]
> Having 6 PRs in 1 will quickly become unwieldy. Please split up future PRs for easier disucssion.

Will be done

**Comment by smcallis** [comment]
Yeah let's split this one up we can probably  submit the others separate from the term indexer one.

**Comment by jmr** [comment]
Closing this since it's been split into #298, #299, #300, #301, #302, and #303.

**Pull request closed without merging**